com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/SyncFlowableStep.java

```diff
@@ -72,21 +76,65 @@ protected ExecutionWrapper createExecutionWrapper(DelegateExecution context) {
         return new ExecutionWrapper(context, stepLogger, clientProvider);
     }
 
-    private void handleException(DelegateExecution context, Throwable t) {
-        t = getWithProperMessage(t);
-        getStepHelper().logException(context, t);
-        throw t instanceof RuntimeException ? (RuntimeException) t : new RuntimeException(t);
+    private void handleException(ExecutionWrapper execution, Exception e) {
+        try {
+            StepPhase stepPhase = StepsUtil.getStepPhase(execution.getContext());
+            if (stepPhase == StepPhase.POLL) {
```

Comment:
This is a bit of a hack. We're using it to avoid processing the exception twice. Can't the processing be done in a single place?
Furthermore, "synchronous" steps do not have polling phases and it is weird for `SyncFlowableStep` to handle this state.

Reply:
Ok. Probably can be improved but I do not have the perfect solution.

I have tried with other approach but without success:
`SuncFlowableStep
protected void onError(ExecutionWrapper execution, Exception e, StepPhase stepPhase) throws Exception {
        processException(e, getStepErrorMessage(execution.getContext()));
    }
`
and
`AsyncFlowableStep
@Override
    protected void onError(ExecutionWrapper execution, Exception e, StepPhase stepPhase) throws Exception {
        if (stepPhase == StepPhase.RETRY) {
            throw e;
        }
        if (stepPhase == StepPhase.POLL) {
            List<AsyncExecution> stepExecutions = getAsyncStepExecutions(execution);
            AsyncExecution stepExecution = getStepExecution(execution, stepExecutions);
            processException(e, stepExecution.getPollingErrorMessage(execution));
        } else {
            processException(e, getStepErrorMessage(execution.getContext()));
        }
    }
`
In this way, the exception stacktrace is lost for `CreateServiceStep.createService`. This gas was present before the change and it is fixed with the provided first option.